storage/src/linear/filebacked.rs

```diff
@@ -106,6 +106,10 @@ impl FileBacked {
             ring: ring.into(),
         })
     }
+
+    pub(crate) fn get_file_size(&self) -> Result<u64, Error> {
+        Ok(self.fd.metadata()?.len())
```

Comment:
The header has a high-water mark which should be used instead of this, as it is certain to be wrong. Lets say a node is allocated at size 2k but we only write 1.9k of data. The file will be shorter than the nodestore header says it is.
```suggestion
        self.header.size
```